extensions/oplogPopulator/allocationStrategy/AllocationStrategy.js

```diff
@@ -5,9 +5,11 @@ class AllocationStrategy {
     /**
      * @constructor
      * @param {Object} params params
+     * @param {PipelineFactory} params.pipelineFactory pipeline factory
      * @param {Logger} params.logger logger object
      */
     constructor(params) {
+        this._pipelineFactory = params.pipelineFactory;
```

Comment:
is the pipeline factory really a part of the strategy?
it seems is is only used as a vessel, to avoid passing an extra parameter to ConnectorManager : but may be cleaner to just keep things separate (lower coupling), and add the parameter where it should be.

Reply:
Indeed, along with the above comment, I also updated the allocation strategy so it does not hold a reference to the pipeline factory, but we pass it to the ConnectorManager, and it does the same, with simpler code...